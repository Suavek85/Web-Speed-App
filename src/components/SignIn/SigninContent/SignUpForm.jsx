import React, { useState } from 'react'
import Input from '../../../utils/Input/Input'
import Button from '../../../utils/Button/Button'
import styles from './SignInForm.scss'
import * as content from '../../../constants/signinContent'

export default function SignUpForm() {

  const [ email, setEmail] = useState('')
  const [ password, setPassword] = useState('')

  const handleSubmit = event => {
    event.preventDefault()
    setEmail('')
    setPassword('')
  }

  const handleEmailChange = event => setEmail(event.target.value)
  const handlePasswordChange = event => setPassword(event.target.value)

  return (
    <div className={ styles.block } >
      <h2>{ content.NO_ACCOUNT }</h2>
      <span>{ content.SIGN_UP_PROMPT }</span>
      <form onSubmit={ handleSubmit } className={ styles.blockForm } >
        <Input 
          label={ content.USERNAME_LABEL }
          name='username' 
          type='text' 
          value={ email } 
          handleChange={ handleEmailChange } 
          required
        />
        <Input 
          label={ content.EMAIL_LABEL }
          name='email' 
          type='email' 
          value={ email } 
          handleChange={ handleEmailChange } 
          required
        />
        <Input 
          label={ content.PASSWORD_LABEL }
          name='password' 
          type='password' 
          value={password} 
          handleChange={ handlePasswordChange } 
          required
        />
        <Input 
          label={ content.CONFIRM_PASSWORD_LABEL }
          name='confirm password' 
          type='password' 
          value={ password } 
          handleChange={ handlePasswordChange } 
          required
        />
        <Button btnStyle='regular' isFormContent />
      </form>
    </div>
  )
}