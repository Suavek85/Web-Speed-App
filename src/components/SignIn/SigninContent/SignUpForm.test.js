import React from 'react'
import { shallow } from 'enzyme'
import SignUpForm from './SignUpForm'

describe('SignUpForm snapshot', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<SignUpForm />)
    expect(wrapper).toMatchSnapshot()
  })

})