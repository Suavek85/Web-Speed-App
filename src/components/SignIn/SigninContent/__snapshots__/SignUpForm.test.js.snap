// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm snapshot should match the snapshot 1`] = `
<div>
  <h2>
    I don't have an account
  </h2>
  <span>
    Sign up with your email and password
  </span>
  <form
    onSubmit={[Function]}
  >
    <Input
      handleChange={[Function]}
      label="Username"
      name="username"
      required={true}
      type="text"
      value=""
    />
    <Input
      handleChange={[Function]}
      label="Email"
      name="email"
      required={true}
      type="email"
      value=""
    />
    <Input
      handleChange={[Function]}
      label="Password"
      name="password"
      required={true}
      type="password"
      value=""
    />
    <Input
      handleChange={[Function]}
      label="Confirm password"
      name="confirm password"
      required={true}
      type="password"
      value=""
    />
    <Button
      btnStyle="regular"
      isFormContent={true}
    />
  </form>
</div>
`;
